src/Translator.zig

```diff
@@ -1412,10 +1518,46 @@ fn transCastExpr(t: *Translator, scope: *Scope, cast: Node.Cast, used: ResultUse
             const sub_expr_node = try t.transExpr(scope, cast.operand, .used);
             return t.maybeSuppressResult(used, sub_expr_node);
         },
+        .int_cast => {
+            const dest_qt = cast.qt;
+            const src_qt = cast.operand.qt(t.tree);
+
+            const operand_expr = try t.transExprCoercing(scope, cast.operand, used);
+
+            const needs_truncate = src_qt.intRankOrder(dest_qt, t.comp).compare(.gt);
+            const needs_bitcast = src_qt.signedness(t.comp) != dest_qt.signedness(t.comp);
+            if (needs_truncate and needs_bitcast) {
+                const as = try ZigTag.as.create(t.arena, .{
+                    .lhs = try t.transTypeIntWidthOf(dest_qt, src_qt.signedness(t.comp) == .signed),
+                    .rhs = try ZigTag.truncate.create(t.arena, operand_expr),
+                });
+                return try ZigTag.bit_cast.create(t.arena, as);
+            } else if (needs_truncate) {
+                return try ZigTag.truncate.create(t.arena, operand_expr);
+            } else if (needs_bitcast) {
+                return try ZigTag.bit_cast.create(t.arena, operand_expr);
+            }
```

Comment:
This isn't quite correct since you might need to extend before bitcasting but it can be left like this for now since it also needs to handle vectors and enums.